tests/test_management_commands.py

```diff
@@ -17,23 +19,124 @@
+        num_available = available_versions.count()
+        expected_msg = (
+            "{pre_message} "
+            "the {index_name} index should have {expected_num} index versions available, "
+            "but instead, there {was_were} {actual_num}".format(
+                pre_message=pre_message,
+                index_name=index_model.name,
+                was_were="were" if num_available > 1 else "was",
+                expected_num=expected_num_available,
+                actual_num=num_available
+            )
+        )
+        self.assertEqual(available_versions.count(), expected_num_available, expected_msg)
+        return available_versions
+
+    def check_last_index_actions(
+            self, index_model, pre_message, num_to_get=1,
+            expected_status=IndexAction.STATUS_COMPLETE, expected_actions=None):
+        """
+        Retrieve the last n IndexActions for the given Index
+        optionally assert the status and action for the retrieved IndexActions
+        :param index_model: the Index model to retrieve actions for
+        :type index_model: django_elastic_migrations.models.Index
+        :param pre_message: description of the action that just was executed
+        :type pre_message: str
+        :param num_to_get: the number of most recent actions to retrieve
+        :type num_to_get: str
+        :param expected_status: the expected status of the IndexAction to assert; by default STATUS_COMPLETE
+        :type expected_status: django_elastic_migrations.models.IndexAction.STATUSES_ALL
+        :param expected_actions: list of action types to assert, in order from oldest to newest
+        :type expected_actions:  [django_elastic_migrations.models.IndexAction.ACTIONS_ALL]
+        :return: IndexActions for the Index
+        :rtype: [django_elastic_migrations.models.IndexAction]
+        """
+        index_actions = index_model.indexaction_set.all().order_by('-pk')[:num_to_get]
+        num_available = index_actions.count()
+        expected_msg = (
+            "{pre_message} "
+            "the {index_name} index should have had {expected_num} {index_action_plural} available, "
+            "but instead, there {was_were} {actual_num}".format(
+                pre_message=pre_message,
+                index_name=index_model.name,
+                expected_num=num_to_get,
+                index_action_plural="IndexAction{}".format(pluralize(num_to_get)),
+                was_were= pluralize(num_available, "was,were"),
+                actual_num=num_available
+            )
+        )
+        self.assertEqual(index_actions.count(), num_to_get, expected_msg)
+
+        # we ordered them in reverse cron to get most recent actions; put them back in cron order
+        index_actions = list(index_actions)
+        index_actions.reverse()
+
+        if expected_status or expected_actions:
+            for num, index_action in enumerate(index_actions, 1):
+                logs = index_action.log[-300:]
```

Comment:
Why the -300 slice?

Reply:
Good question, the idea was that if something fails in a test in CI, it would be easier to see the last part of the stack trace (rather than the whole log). I ended up raising the exception, though, so this part probably isn't actually relevant anymore; I'll update.

Reply:
updated in 481968bb9